File: lab1-frontend/src/app/common/task-dto.ts
import {ProjectDto} from "./project-dto";
import {User} from "./user";
import {Status} from "../enums/status";

export class TaskDto {
  constructor(
    public id: number,
    public project: ProjectDto,           // Reference to ProjectDto
    public assignedEmployee: User,     // Reference to UserDto
    public taskSubject: string,
    public taskDetail: string,
    public taskStatus: Status,            // Assuming Status is an enum
    public assignedDate: Date              // Using Date type for assigned date
  ) {}
}

File: src/main/java/org/foo/dto/TaskDto.java
package org.foo.dto;


import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.foo.enums.Status;

import java.time.LocalDate;

@NoArgsConstructor
@Data
@AllArgsConstructor
public class TaskDto {

  private Long id;

  private ProjectDto project;

  private UserDto assignedEmployee;

  private String taskSubject;

  private String taskDetail;

  private Status taskStatus;
  private LocalDate assignedDate;



}
